src/Assessment/Processor/ProcessDatasets.cs

```diff
@@ -1243,6 +1249,277 @@ private void UpdateIndividualWebAppPaaSCosts(List<WebApp_PaaS> WebApp_PaaS_List)
+            Dictionary<string, HashSet<string>> EnvironmentToUniqueIdMap = new Dictionary<string, HashSet<string>>();
+            EnvironmentToUniqueIdMap["Dev"] = new HashSet<string>();
+            EnvironmentToUniqueIdMap["Prod"] = new HashSet<string>();
+
+            foreach (var sqlinstance in SQL_IaaS_Server_Rehost_Perf_List)
+            {
+                if (sqlinstance.Environment.Equals("Dev"))
+                {
+                    if (!EnvironmentCountDictionary.ContainsKey("Dev"))
+                    {
+                        EnvironmentCountDictionary.Add("Dev", 0);
+                    }
+                    if (!EnvironmentToUniqueIdMap["Dev"].Contains(sqlinstance.MachineId))
+                        EnvironmentToUniqueIdMap["Dev"].Add(sqlinstance.MachineId);
+                }
+                else
+                {
+                    if (!EnvironmentCountDictionary.ContainsKey("Prod"))
+                    {
+                        EnvironmentCountDictionary.Add("Prod", 0);
+                    }
+                    if (!EnvironmentToUniqueIdMap["Prod"].Contains(sqlinstance.MachineId))
+                        EnvironmentToUniqueIdMap["Prod"].Add(sqlinstance.MachineId);
+                }
+                EnvironmentCountDictionary[sqlinstance.Environment] = +1;
+            }
+
+            foreach (var sqlinstance in SQL_IaaS_Instance_Rehost_Perf_List)
+            {
+                if (sqlinstance.Environment.Equals("Dev"))
+                {
+                    if (!EnvironmentCountDictionary.ContainsKey("Dev"))
+                    {
+                        EnvironmentCountDictionary.Add("Dev", 0);
+                    }
+                    if (!EnvironmentToUniqueIdMap["Dev"].Contains(sqlinstance.MachineId))
+                        EnvironmentToUniqueIdMap["Dev"].Add(sqlinstance.MachineId);
+                }
+                else
+                {
+                    if (!EnvironmentCountDictionary.ContainsKey("Prod"))
+                    {
+                        EnvironmentCountDictionary.Add("Prod", 0);
+                    }
+                    if (!EnvironmentToUniqueIdMap["Prod"].Contains(sqlinstance.MachineId))
+                        EnvironmentToUniqueIdMap["Prod"].Add(sqlinstance.MachineId);
+                }
+                EnvironmentCountDictionary[sqlinstance.Environment] = +1;
+            }
+
+            SQLIaaSObj.DevMachinesCountTarget = EnvironmentCountDictionary["Dev"];
+            SQLIaaSObj.ProdMachinesCountTarget = EnvironmentCountDictionary["Prod"];
+            SQLIaaSObj.DevUniqueMachinesSource = EnvironmentToUniqueIdMap["Dev"].Count;
+            SQLIaaSObj.ProdUniqueMachinesSource = EnvironmentToUniqueIdMap["Prod"].Count;
+            SQLIaaSObj.EnvironmentToUniqueIdMap = EnvironmentToUniqueIdMap;
+            SQLIaaSObj.EnvironmentCountDictionary = EnvironmentCountDictionary;
+            return SQLIaaSObj;

```

Comment:
P0: EnvironentToUniqueIdMap is only required for Backup and Management server cost distinction. If that logic can be moved to AzureIaaSCalculator, can we reduce the number of classes we require.

Reply:
we can reduce the number of objects defined in the respective class, definitely, but not an entire class.